### PR TITLE
backport 1.16: ci: fix docs tag build

### PR DESCRIPTION
**Pull request by Shikugawa**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -281,6 +282,43 @@ stages:
...
+              sshPublicKey: "$(DocsPublicKey)"
+              sshPassphrase: "$(SshDeployKeyPassphrase)"
+              sshKeySecureFile: "$(DocsPrivateKey)"
+
+          - script: docs/publish.sh
+            displayName: "Publish to GitHub"
+            workingDirectory: $(Build.SourcesDirectory)
+            env:
+              AZP_BRANCH: $(Build.SourceBranch)
+
```
Comment (lizan) [inline .azure-pipelines/pipelines.yml:321]: This part isn't needed in 1.16 and prior versions.

**Comment by Shikugawa** [comment]
@moderation Could you approve this?

**Pull request merged**